lib/cloudregister/registerutils.py

```diff
@@ -836,6 +917,199 @@ def clean_bashrc_local(env_vars):
     return bashrc_local_new_lines, modified, False, False
 
 
+# ----------------------------------------------------------------------------
+def clean_registries_conf():
+    """Clean up the registry content from Podman and/or Docker config files."""
+    clean_registries_conf_podman()
+    clean_registries_conf_docker()
+
+
+# ----------------------------------------------------------------------------
+def clean_registries_conf_podman():
+    """Clean up the registry content from the REGISTRIES_CONF_PATH file."""
+    if not os.path.exists(REGISTRIES_CONF_PATH):
+        return True
+
+    registries_conf, failed = get_registry_conf_file(
+        REGISTRIES_CONF_PATH, 'podman'
+    )
+    if failed:
+        return False
+
+    if not registries_conf and not failed:
+        logging.info(
+            'Either %s is empty or it has been backed up' %
+            REGISTRIES_CONF_PATH
+        )
+        return True
+
+    modified = False
+    unqualified_search_reg = registries_conf.get(
+        'unqualified-search-registries', []
+    )
+    pub_registry_fqdn = 'registry.suse.com'
+    smt = get_smt_from_store(__get_registered_smt_file_path())
+    if unqualified_search_reg:
+        if smt and smt.get_registry_FQDN():
+            private_registry_fqdn = smt.get_registry_FQDN()
+            if private_registry_fqdn in unqualified_search_reg:
+                unqualified_search_reg.pop(
+                    unqualified_search_reg.index(private_registry_fqdn)
+                )
+                modified = True
+            if pub_registry_fqdn in unqualified_search_reg:
+                unqualified_search_reg.pop(
+                    unqualified_search_reg.index(pub_registry_fqdn)
+                )
+                modified = True
```

Comment:
no, because the 

```python

            if pub_registry_fqdn in unqualified_search_reg:
                unqualified_search_reg.pop(
                    unqualified_search_reg.index(pub_registry_fqdn)
                )
                modified = True
                
````

was outside that check, but given that there is a [loop](https://github.com/SUSE-Enceladus/cloud-regionsrv-client/pull/165/files#diff-84460adc47c95f16ae220daac1882a7bc19b595d1ba01650e286b666ef14f2a4R967)

I decided to check in the loop for `registry.suse.com` instead of loop + _another_ operation to check

Reply:
But this code does loop + another operation. If we have the fqdn for the update infrastructure we check to remove it and we check to remove registry.suse.com, if we do not have the fqdn of the update server we check for the domain name of the update infrastructure **and** we check for registry.suse.com. Meaning we check for registry.suse.com in the `if` and the `else` block. Meaning we do the same operation in any condition and as such that operation should then happen unconditionally.

Reply:
yes, the operation happening unconditionally was as it was, thought to piggy back on the for-loop

will change it in the refactor, as you mentioned